### DISCUSSION TITLE
Limit Switch triggered on G28 G91 Z0

### DISCUSSION THREAD

**Discussion by HubertJH** [winder/Universal-G-Code-Sender#2464] [Help] [answered]
I have hall sensors installed as limit switches on all three axes, homing works as expected.
When I am running g-code created in Fusion360 the z-axis is first moved to 0 (G28 G91 Z0); however, this triggers the limit switch at a few mm/100 before 0. I have played with the Homing switch pull-off distance with no success; the limit switch triggers regardless of the distance (even at 4mm).
While I can edit the g-code to overcome this issue I was wondering whether I am getting something wrong with the limit switch settings; from the description in grbl I concluded that the pull-off distance would prevent the machine from actually trying to move to 0. 
Any help would be very much appreciated.

**Comment by breiler** [comment]
`G28` will return to a predefined position. This will be at `0, 0, 0` by default. If you are using homing there is normally a pull off distance from the limit switches (default 1 mm on GRBL). So if you go to the machine coordinate `0, 0, 0` you will likely (just barely) trigger your limit switches.

Try to execute `G28.1` after you've homed your machine. This will set the G28 coordinate system to your current position. You can check it by issuing `$#`. 

You should now be able to use `G28 G91 Z0`.

**Reply by HubertJH** [reply]
Thanks for your swift response. Sounds like this should do the trick. Let me try that and I will report back.

**Comment by HubertJH** [comment, marked as answer]
Run a quick test job and the G28.1 command executed after the homing sequence the job ran just fine. So again thanks a lot for you help in this matter.
I was wondering though whether UGS could execute G28.1 automatically after completion of the homing sequence, maybe as an option.
